doc/source/reference/random/index.rst

```diff
@@ -0,0 +1,237 @@
+.. _numpyrandom:
+
+.. currentmodule:: numpy.random
+
+numpy.random
+============
+
+Numpy's random number routines produce psuedo random numbers using
+combinations of generators and distributions:
+
+* Generators: Objects that generate random numbers. These are typically
+  unsigned integer words filled with sequences of either 32 or 64 random bits.
+* Distributions: Objects that transform sequences of random bits from a
+  generator into sequences of numbers that follow a specific probability
+  distribution (such as uniform, Normal or Binomial) within a specified
+  interval.
+
+Since Numpy version 1.17.0 the `RandomGenerator` can be initialized with a
+number of different Random Number Generators (RNG)s, and exposes many
+different probability distributions. See
+`NEP 19 <https://www.numpy.org/neps/nep-0019-rng-policy.html>`_ for context on
+the updated random Numpy number routines. The legacy `mtrand.RandomState`
+random number routines are still available, but limited to a single generator.
+
+For convenience and backward compatibility, a single `mtrand.RandomState`
+instance's methods are imported into the numpy.random namespace, see
+:ref:`legacy` for the complete list.
+
+Quick Start
+-----------
+
+By default, `RandomGenerator` uses normals provided by
+`xoroshiro128.Xoroshiro128` which will be faster than the legacy methods in
+`mtrand.RandomState`
+
+.. code-block:: python
+
+  # As replacement for numpy.random.RandomState
+  from numpy import random
+  random.standard_normal()
+
+`RandomGenerator` can be used as a direct replacement for
+`~RandomState`, although the random values are generated by
+`~xoroshiro128.Xoroshiro128`. The `RandomGenerator` holds an instance of a RNG.
+It is accessable as ``gen.brng``.
+
+.. code-block:: python
+
+  # As replacement for RandomState()
+  from numpy.random import RandomGenerator
+  rg = RandomGenerator()
+  rg.standard_normal()
+
+
+Seeds can be passed to any of the basic RNGs. Here `mt19937.MT19937` is used
+and the ``RandomGenerator`` is accessed via the attribute `mt19937.MT19937.
+generator`.
+
+.. code-block:: python
+
+  from numpy.random import MT19937
+  rg = MT19937(12345).generator
+  rg.standard_normal()
+
+
+Introduction
+------------
+RandomGen takes a different approach to producing random numbers from the
+:class:`numpy.random.RandomState` object.  Random number generation is
+separated into two components, a basic RNG and a random generator.
+
+The basic RNG has a limited set of responsibilities. It manages the
+underlying RNG state and provides functions to produce random doubles and
+random unsigned 32- and 64-bit values. The basic random generator also handles
+all seeding since this varies when using alternative basic RNGs.
+
+The `random generator <~RandomGenerator>` takes the
```

Comment:
@bashtage The fact that this concept is called a 'random generator' is, for me at least, a possible source of confusion since it is named similar to 'random number generator (RNG)'. In C++, for example, this concept is simply called a 'distribution' which is perhaps better because it is easy to remember, but also not entirely correct because it is really a 'transformation' from the brng distribution to a more useful distribution.

I don't have a suggestion on how the naming might be improved, but I'm now wondering if the change at the top of this file to mention distributions was the correct update to make.

Reply:
We have renamed the main class to simply Generator, so that it is numpy.random.Generator.  The core PRNGs are now called bit generators, since this is what they do.  I think these are better names, but it is never perfect when naming things.